gufe/protocols/protocoldag.py

```diff
@@ -0,0 +1,59 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+import abc
+from typing import Iterable, List, Dict
+
+from .protocolunit import ProtocolUnit
+from .results import ProtocolDAGResult
+
+
+class ProtocolDAG:
+    """An executable directed, acyclic graph (DAG) composed of `ProtocolUnit`s
+    with dependencies specified.
+
+    This is the unit of execution passed to an alchemical `Scheduler`.
+    A `ProtocolDAG` yields a `ProtocolResult`, which can be placed in a `ResultStore`.
+
+    """
+
+    def __init__(
+            self,
+            name: str,
+            protocol_units: Iterable[ProtocolUnit]
+        ):
+        """Create a new `ProtocolDAG`.
+
+        Parameters
+        ----------
+        name : str
+            Name of the `Protocol` that generated this `ProtocolDAG`.
+        protocol_units : Iterable[ProtocolUnit]
+            The `ProtocolUnit`s, with dependencies set, to include in the `ProtocolDAG`.
+
+        """
+        self._name = name
+        self._protocol_units = tuple(protocol_units)
+
+    @property
+    def name(self):
+        return self._name
+
+    def to_dask(self):
+        """Produce a `dask`-executable DAG from this `ProtocolDAG` as a `dask.Delayed` object.
+
+        """
+        ...
```

Comment:
Can we keep scheduler specific out of this?  I.e. can we put this logic in `DaskScheduler.submit(thing)`

Reply:
Unless this is literally the object that dask consumes, in which case that's ok.  Not sure if we're implementing the scheduler interfaces or not.

Reply:
This was more of a stub for an idea, where we can produce fully-executable `dask` graphs that can be run with `dask`'s own schedulers (local and `distributed`). I can remove it from this PR and consider it again later.